sportcomponentlib/src/main/java/com/sportsworld/sportcomponentlib/ReusableSportUILib.kt

```diff
@@ -30,46 +26,74 @@ fun TopBarLib() {
 
 @Composable
 fun DisplayScreenSetup(viewModel: SportFeatureViewModelLib = viewModel()) {
-    MainScreen(viewModel.sharedFlow)
+    MainScreen(viewModel)
 }
 
 @Composable
-//Todo: read up on side effects in JetPack Compose with coroutines - would not put this code into production until I understood these ideas better
-fun MainScreen(sharedFlow: SharedFlow<List<Sport>>) {
+fun RefreshButtonLib(viewModel: SportFeatureViewModelLib = viewModel()) {
+    val scope = rememberCoroutineScope()
+    // In practice would pull the last sport displayed from Shared Prefs.
+    var currentSport = Sport.createMockedSports()[0]
 
-    val messages = remember { mutableStateListOf<List<Sport>>() }
+    val getNewSportOnClick: () -> Unit = {
+        scope.launch {
+            try {
 
-    LaunchedEffect(key1 = Unit) {
-        sharedFlow.collect {
-            messages.add(0, it)
-            // Todo: Learn more about recomposition so I can find a better way to achieve this
-            messages.removeRange(1,messages.size)
+                // try and collect the value from the flow and then print it out
+                viewModel.loadNews()
+
+                val listOfSports = viewModel.uiState.value
+
+                Log.d("katie", currentSport.name) // testing current sport is getting updated
+                // Don't want to display same sport twice in a row
+                if(currentSport == listOfSports[0])
+                    currentSport = listOfSports[1]
+                else
+                    currentSport = listOfSports[0]
+
+            } catch (ex: Exception) {
+                println("Coroutine to get current list of sports cancelled")
+            }
         }
     }
-    LazyColumn {
-        items(messages) {
-            val sport = it[0].name
-            val description = it[0].description
+
+    RefreshIcon(onClick = getNewSportOnClick) {
```

Comment:
This is not building for me here

Reply:
Hopefully now that RefreshIcon has been added the build will proceed.